packages/cosmos-tigergraph/src/index.ts

```diff
@@ -0,0 +1,185 @@
+import { InputNode, InputLink } from '@cosmograph/cosmos'
+
+export class TigerGraphConnection<N extends InputNode, L extends InputLink> {
+  private host: string
+  private graphname: string
+  private username: string
+  private password: string
+  private token: string
+
+  private constructor (host: string, graphname: string, username: string, password: string, token?: string) {
+    this.host = host
+    this.graphname = graphname
+    this.username = username
+    this.password = password
+    this.token = token ?? ''
+  }
+
+  public async generateToken () {
```

Comment:
@GenericP3rson We are declaring the functions as `async` but we are not using the traditional Promise syntax function inside them instead of `await`. Is there a particular reason for that?

Reply:
@rokotyan There's no particular reason; we can switch it to one or the other!

Reply:
@GenericP3rson Great, do you think you can do it?

Reply:
Sure, @rokotyan! Do you prefer one method over the other or would either work?